hazelcast-client/src/main/java/com/hazelcast/client/proxy/PartitionSpecificClientProxy.java

```diff
@@ -43,21 +41,28 @@ protected void onInitialize() {
         partitionId = getContext().getPartitionService().getPartitionId(partitionKey);
     }
 
-    protected ClientMessage invokeOnPartition(ClientMessage req) {
-        return invokeOnPartition(req, partitionId);
+    protected ClientMessage invokeOnPartition(ClientMessage request) {
+        return invokeOnPartition(request, partitionId);
+    }
+
+    protected <T> T invokeOnPartition(ClientMessage request, ClientMessageDecoder decoder) {
```

Comment:
I've always been puzzled when reading Hazelcast's codebase by duplicate logic in both sync and async implementations of the same method. Is the reason why this is the case so that we avoid an extra method call incurred if sync delegates to async, assuming the implementation is simple enough?